crates/biome_js_analyze/src/analyzers/nursery/no_empty_block_statements.rs

```diff
@@ -0,0 +1,176 @@
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// function foo () {}
+    ///
+    /// const foo = () => {}
+    ///
+    /// function fooWithNestedEmptyBlock() {
+    ///     let a = 1;
+    ///     function shouldFail(){}
+    ///     return a
+    ///  }
+    ///
+    /// const fooWithNestedEmptyBlock = () => {
+    ///     let a = 1;
+    ///     const shouldFail = () => {}
+    ///     return a
+    ///  }
+    /// let someVar;
+    /// if (someVar) {
+    /// }
+    ///
+    /// while (someVar) {
+    /// }
+    ///
+    /// switch(someVar) {
+    /// }
+    /// try {
+    ///     doSomething();
+    /// } catch(ex) {
+    ///
+    /// } finally {
+    ///
+    /// }
+    ///
+    // class Foo {
+    //   static {}
+    // }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// function foo () {let a;}
+    ///
+    /// const foo = () => {let a;}
+    ///
+    /// function fooWithComment() {
+    ///   // should work
+    /// }
+    ///
+    /// const barWithComment = () => {
+    ///   // should work
+    /// }
+    ///
+    /// function fooWithMultilineComment() {
+    ///   /**
+    ///    * this should also work
+    ///    */
+    /// }
+    ///
+    /// const barWithMultilineComment = () => {
+    ///   /**
+    ///    * this should also work
+    ///    */
+    /// }
+    ///
+    ///
+    /// if (foo) {
+    ///   // empty
+    /// }
+    ///
+    /// while (foo) {
+    ///   /* empty */
+    /// }
+    ///
+    /// try {
+    ///   doSomething();
+    /// } catch (ex) {
+    ///   // continue regardless of error
+    /// }
+    ///
+    /// try {
+    ///   doSomething();
+    /// } finally {
+    ///   /* continue regardless of error */
+    /// }
+    ///
+    /// class Foo {
+    ///   static {
+    ///       bar();
+    ///   }
+    /// }
+    ///
+    /// class Foo {
+    ///   static {
+    ///       // comment
+    ///   }
+    /// }
+    /// ```
+    ///
+    pub(crate) NoEmptyBlockStatements {
+        version: "next",
+        name: "noEmptyBlockStatements",
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub(crate) Query = JsBlockStatement | JsFunctionBody | JsStaticInitializationBlockClassMember | JsSwitchStatement
+}
+
+impl Rule for NoEmptyBlockStatements {
+    type Query = Ast<Query>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let query = ctx.query();
+        let is_empty = is_empty(query);
+        let has_comments = query.syntax().has_comments_descendants();
+        let text_range = query.syntax().text_range();
+
+        if is_empty && !has_comments {
+            return Some(text_range);
+        }
+
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                state,
```

Comment:
`query.range()` internally calls `query.syntax().text_trimmed_range()`.

```suggestion
                query.range(),
```